core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp

```diff
@@ -642,13 +682,23 @@ struct CudaReductionsFunctor<FunctorType, ArgTag, false, false> {
       const bool skip_vector,                  // Skip threads if Kokkos vector lanes are not part of the reduction
       const int width)                         // How much of the warp participates
   {
+#ifdef KOKKOS_IMPL_CUDA_SYNCWARP_NEEDS_MASK
     unsigned mask = width==32?0xffffffff:((1<<width)-1)<<((threadIdx.y*blockDim.x+threadIdx.x)%(32/width))*width;
+#endif
     const int lane_id = (threadIdx.y*blockDim.x+threadIdx.x)%32;
     for(int delta=skip_vector?blockDim.x:1; delta<width; delta*=2) {
-      if(lane_id + delta<32)
+      if(lane_id + delta<32) {
         ValueJoin::join( functor , value, value+delta);
+      }
+#ifdef KOKKOS_IMPL_CUDA_SYNCWARP_NEEDS_MASK
       KOKKOS_IMPL_CUDA_SYNCWARP_MASK(mask);
+#else
+      KOKKOS_IMPL_CUDA_SYNCWARP_MASK;
+#endif
     }
+#ifdef KOKKOS_IMPL_CUDA_SYNCWARP_NEEDS_MASK
+    fcn_to_silence_unused_var_warnings(mask);
```

Comment:
This `fcn_to_silence_unused_var_warnings` shouldn't be needed anymore?

Reply:
I needed this one call after the rest of the changes to silence the warning here but I'm not sure why; the other spots I commented out I'll clean up.